### ISSUE TITLE
Need more developer documentation on creating tests

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#10410]
Developers need more documentation on how to create tests and how the various types of tests work.  test-suite/README.md only mentions tests that are in "output" (which compares output to expected) and "unit-tests" directories--without an explanation of how those tests work.  But there's no explanation of how, say, "interactive" tests work (for which I haven't yet figured out how to cause a test failure!).

test-suite has about 20 subdirectories.  Putting a test in one of them causes specific behavior (e.g. "output") which others appear simply to group related tests together.

(I expect we'd like to see many improvements in the test suite, but that's a different topic.)

**Comment by Zimmi48** [comment]
Indeed, it is my understanding that a lot of sub-directories behave the same (just run coqc then coqchk). The separation into components is nice to have but these sub-directories should probably all be gathered beyond a single root.
Exceptions that I know of are `complexity`, `coq-makefile` (which IIUC behaves mostly like `misc`), `interactive`, `misc`, `output`, `output-modulo-time` and `unit-tests`, but I wouldn't be able to describe most of them. To solve this issue, we will need contributions from various people (for instance @herbelin for `complexity` and @gares for `interactive`).

**Comment by ejgallego** [comment]
#13364 will include a significant doc update about this, hopefully enough to close this.
